src/sbd-common.c

```diff
@@ -662,6 +662,112 @@ static void sbd_memlock(int stackgrowK, int heapgrowK)
 #endif
 }
 
+static int get_realtime_budget(void)
+{
+    FILE *f;
+    char fname[PATH_MAX];
+    int res = -1, lnum = 0;
+    char *cgroup = NULL, *namespecs = NULL;
+
+    snprintf(fname, PATH_MAX, "/proc/%jd/cgroup", (intmax_t)getpid());
+    f = fopen(fname, "rt");
+    if (f == NULL) {
+        cl_log(LOG_WARNING, "Can't open cgroup file for pid=%jd",
+                            (intmax_t)getpid());
+        goto exit_res;
+    }
+    while( fscanf(f, "%d:%m[^:]:%m[^\n]", &lnum,  &namespecs, &cgroup) !=EOF ) {
+        if (namespecs && strstr(namespecs, "cpuacct")) {
+            free(namespecs);
+            break;
+        }
+        if (cgroup) {
+            free(cgroup);
+            cgroup = NULL;
+        }
+        if (namespecs) {
+            free(namespecs);
+            namespecs = NULL;
+        }
+    }
+    fclose(f);
+    if (cgroup == NULL) {
+        cl_log(LOG_WARNING, "Failed getting cgroup for pid=%jd",
+                            (intmax_t)getpid());
+        goto exit_res;
+    }
+    snprintf(fname, PATH_MAX, "/sys/fs/cgroup/cpu/%s/cpu.rt_runtime_us",
+                              cgroup);
```

Comment:
Won't cgroup always start with /, so you can use cpu%s?

Reply:
On the kernels I've tried yes. When I realized I still left it like this to be able to cope with an empty-string case or without leading /.
But it is probably OK to remove it.
What I'm not 100% sure of is if it should rather be cpu or cpuacct.
In all cases I've seen they were both links to cpu,cpuacct or cpuacct,cpu.
But does it have to be like that? I've chosen to search for cpuacct in /proc/$pid/cgroup just because it is easier to implement (cpu wouldn't be unique because of e.g. cpuset - would have to go through parsing for comma and stuff).

Reply:
@wenningerk AFAIK cpuacct was older version later replaced by cpu. I don't think this will change because cgroup v1 should be replaced by v2 (https://fedoraproject.org/wiki/Changes/CGroupsV2). I'm personally going to install Fedora rahwhide quite soon to find out what has changed and what need to be changed.

Reply:
In Rawhide, `cgroup2` pseudo-filesystem gets mounted to
`/sys/fs/cgroup/unified` (and systemd is compiled with `default-hierarchy=hybrid`).

Reply:
... and CONFIG_RT_GROUP_SCHED seems to be disabled.
Otherwise the old structure is still there including ...:cpu,cpuacct:/ in /proc/$pid/cgroup except cpu.rt_runtime_us of course.
But then nothing has to be done about rt-budget anyway probably and the code would be quite silent (1 debug message).

Reply:
@wenningerk Disabled CONFIG_RT_GROUP_SCHED is standard for Fedora (RHEL differs in this aspect). The thing is to really to find out how new unified hierarchy differs (not hybrid one, because hybrid = cgroup v1 + v2)

Reply:
@jfriesse as said rest looked quite familiar - guess only way to find out is building a kernel with CONFIG_RT_GROUP_SCHED enabled ...

Reply:
@wenningerk Yep, looks so.